src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(new Random(4000).nextLong());
+        } catch (InterruptedException e) {
+            System.out.println("Interrupted for " + userId);
+        }
+        UserData cashedUserData = cache.get(userId);
+        if (cashedUserData != null) {
+            return cashedUserData;
+        }
+        UserData userData = new UserData(userId, "Details for " + userId);
```

Comment:
The hardcoded string 'Details for ' concatenated with `userId` suggests that user details are not dynamically fetched. If this is a placeholder for actual data retrieval, consider adding a comment or a TODO to indicate that real user data fetching needs to be implemented.